### DISCUSSION TITLE
[Feature] Utilities Duplicates - Keep Album Selection and Favorite Status (for Trashing and Stacking)

### DISCUSSION THREAD

**Discussion by frederikb96** [immich-app/immich#14751] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

## Problem
It would be awesome to automatically add the photo, which is kept, to the album of the photo, which is removed via deduplication process, such that one is not loosing the photo in the album. Same for favorite status of an image which is removed.

I guess this is actually a quite **essential functionality** since right now, one **accidentally removes images from albums** if one is not checking every time the details and only removes images which are in the same album. Use case is quite common for photo collections for birthdays or calendars etc, where one might have uploaded a low quality messenger image, and the real image is part of the holiday album. At some point, I realized that my collections dont have pictures anymore :D

## Example

Example:
Img1, low quality and in album A
Img2, high quality and in album B
-> Remove Img1, and add Img2 to album A

### Edge Case for multiple images:
We always want to keep all albums preserved.

Example, keeping one image:
Img1, low quality in album A
Img2, low quality in album B
Img3, high quality in album C
-> Remove Img1 and Img2, and add Img3 to album A and B

Example, keeping multiple images:
Img1, low quality in album A
Img2, high quality in album B
Img3, high quality in album C
-> Remove Img1, and add Img2 and Img3 to album A

I guess these edge cases with multiple photos being kept are not so important, since they will rarely occur in combinations with photos being in different albums, but it is better to just add those photos to the album than removing it and loosing the album state :)

---

## So the rule for implementation could be
1. When user has made selection of which images to keep and which to remove, we get an array of albums where the removed assets are included in, and also a flag if at least one of the removed images is a favorite
2. And then we add each kept image to the array of albums. And also mark it as favorite, if favorit flag is set.

Optional for stacking images:
Just get list of all albums of all stacked photos, and add each photo to each album, such that the stack appears in each album. Right now, when stacking photos, the stack appears in timeline, but in album its not visible, since only one of the pictures is part of the album. Here we don't have problem with edge cases, since no photos are deleted, and simply all are stacked.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#10262

**Reply by frederikb96** [reply]
Ah perfect :+1: 

Btw: Sorry for not finding it, was hidden somewhere low in discussions when searching for duplicate and album
